test/e2e/dns.go

```diff
@@ -0,0 +1,409 @@
+						{
+							Name: "resolv-conf",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/etc/resolv.conf",
+									Type: &hpt,
+								},
+							},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyOnFailure,
+				},
+			},
+		},
+	}
+	_, err := cli.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+	return err
+}
+
+func deleteResolvConfJob(ctx context.Context, cli kubernetes.Interface, nodeName string, namespace string) error {
+	dpb := metav1.DeletePropagationBackground
+	return cli.BatchV1().Jobs(namespace).Delete(ctx, resolvConfJobName(nodeName), metav1.DeleteOptions{
+		PropagationPolicy: &dpb,
+	})
+}
+
+func verifyResolvConf(ctx context.Context, cli kubernetes.Interface, nodeName string, namespace string, nodeIp string) error {
+	jobName := resolvConfJobName(nodeName)
+	podList, err := cli.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("job-name=%s", jobName),
+	})
+	if err != nil {
+		return err
+	}
+	if len(podList.Items) != 1 {
+		return fmt.Errorf("found %v Pods associated with the Job, but there should be exactly 1", len(podList.Items))
+	}
+
+	podName := podList.Items[0].ObjectMeta.Name
+	tailLines := int64(10)
+	podLogOptions := &corev1.PodLogOptions{
+		Container: resolvConfContainerName,
+		Follow:    false,
+		TailLines: &tailLines,
+	}
+	podLogRequest := cli.CoreV1().Pods(namespace).GetLogs(podName, podLogOptions)
+	stream, err := podLogRequest.Stream(ctx)
+	if err != nil {
+		return err
+	}
+	defer stream.Close()
+
+	logs := []string{}
+	for {
+		buf := make([]byte, 2000)
+		numBytes, err := stream.Read(buf)
+		if numBytes == 0 {
+			break
+		}
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		logs = append(logs, string(buf[:numBytes]))
+	}
+
+	log := strings.Join(logs, "\n")
+	if log == "" {
+		return fmt.Errorf("pod didn't have any logs")
+	}
+
+	nameserverLine := nameserverRegex.FindString(log)
+	if nameserverLine == "" {
+		return fmt.Errorf("didn't find nameserver in resolv.conf")
+	}
+
+	_, nameserverIp, _ := strings.Cut(nameserverLine, " ")
+	if nameserverIp != nodeIp {
+		return fmt.Errorf("nameserver specified in resolv.conf does not match node's IP address")
+	}
+
+	return nil
+}
+
+func toggleAcceleratedNetworking(ctx context.Context, interfaces network.InterfacesClient, clusterResourceGroup string, nodeName string, enabled bool) error {
+	nic, err := interfaces.Get(ctx, clusterResourceGroup, nicName(nodeName), "")
+	if err != nil {
+		return err
+	}
+
+	nic.EnableAcceleratedNetworking = to.BoolPtr(enabled)
+	err = clients.Interfaces.CreateOrUpdateAndWait(ctx, clusterResourceGroup, nicName(nodeName), nic)
+	return err
+}
+
+func nodesReady(ctx context.Context, cli kubernetes.Interface) error {
+	nodeList, err := cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, node := range nodeList.Items {
```

Comment:
Do we need to check readiness on all nodes or just the ones we're testing (worker(s))?